### DISCUSSION TITLE
Issues when running FLORIS 4.1.1

### DISCUSSION THREAD

**Discussion by prith-gs** [NatLabRockies/floris#950] [General]
Hello,

I'm trying to run example 01 with FLORIS 4.1.1 and I don't get this error in FLORIS 3.5 and I'm not sure why I get this error.

Here's the code I'm running where I am reading the input yaml file using FlorisModel function in line 9 and I'm using a function I found online in line 4 and 5.
![image](https://github.com/user-attachments/assets/0611306b-942f-49a4-b70e-d73f9309849f)

Line 4 and 5 executes properly and has no issues in reading the yaml file and I can see the variable 'data' as well
![image](https://github.com/user-attachments/assets/bc85e0b0-94de-4d08-bac9-0d7ead5790e4)

Here's what's stored in the variable data
![image](https://github.com/user-attachments/assets/5b85766b-9c78-43f0-bcc1-1a4ee7f8ac85)

But line 9 gives me the following error and I'm not clear why I'm getting this error
![image](https://github.com/user-attachments/assets/61b0cee3-00ba-4d9d-9944-89ea28406b58)

Any help is much appreciated.
Thanks!

**Comment by HDW7812** [comment]
Hi, I'm also experiencing this problem, although I don't know exactly what's causing it, but I have a feeling it's a conflict with a previously installed package, so the turbine_library_path is incorrect. I create a new envs (without pip install floris), then run the demo, it works. I hope it can help you!
If you are in a hurry to use it, you can try this method. Wish someone could answer you more professionally.

**Reply by prith-gs** [reply]
Appreciate the response! but unfortunately it still doesn't work for me. I'm not sure If I'm missing something else.

**Reply by HDW7812** [reply]
I apologize that I neglected to mention the python version I created the new environment with, my original version was 3.9 subsequently changed to 3.10 and it worked. Like @edwards1988  said.

**Reply by prith-gs** [reply]
No issues. It's working now. Thanks!

**Comment by misi9170** [comment]
Hi @prith-gs , 

If you are using this within a conda environment, could you share the list of packages you have installed? As @HDW7812 pointed out, this could be something to do with package versions in your environment, especially if you are using the same environment you were running v3.5 with.

As also pointed out by @HDW7812 , starting with a fresh conda environment and a fresh pip install of FLORIS may solve your issue. 

I'm going to convert this discussion to an issue in the meantime.

Cheers,
Misha

**Comment by edwards1988** [comment]
I create a new environment in anaconda, and reinstall floris. The issue is still open. However, My colleague tries the python 3.12.4 on pycharm and the issue solved. I doubt wether the version of python leads to this issue.

**Reply by prith-gs** [reply]
This worked. Thank you!

**Reply by edwards1988** [reply]
Now I tried it on anaconda. My original environment (base) is based on python 3.8.18.  I created an new virtual environment called 'floris4_env' with python 3.8 which is the latest version that my anaconda version can provide.  Afterwards, I installed spyder with a recent version 5.5.1, and then I upgraded python version to 3.12 successfully.  The last step is installing floris 4.  Remember the steps are important in order to keep the environment clean.

**Comment by misi9170** [comment]
Now that we have an issue open for this #951 , I'm going to close this discussion. Please refer to #951.
